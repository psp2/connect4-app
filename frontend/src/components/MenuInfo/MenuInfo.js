import React from "react";
import "./MenuInfo.scss";

function MenuInfo(props) {
  const tutorialText =
    "Here's how to play Connect 4 ... write more once implementation is complete!";
  const aboutGameText =
    "This Connect4 app and implementation was designed by Prashant Pokhriyal and Vivek Sriramineni (based on Python scripts written by Prashant Pokhriyal) for a UIUC CS242 class project.\n \n Connect Four belongs to Hasbro, Inc., and we do not aim to profit off of this game. \n\n With that being said, sit back and enjoy the game!";

  if (props.displayType === "tutorial") {
    return (
      <div className="Tutorial">
        <div className="TutorialText">{tutorialText}</div>
        <div className="CloseTutorial" onClick={props.closeDisplay}>
          <div className="ButtonCaption">Close</div>
        </div>
      </div>
    );
  } else {
    return (
      <div className="AboutGame">
        <div className="AboutGameText">{aboutGameText}</div>
        <div className="CloseAboutGame" onClick={props.closeDisplay}>
          <div className="ButtonCaption">Close</div>
        </div>
      </div>
    );
  }
}

export default MenuInfo;
