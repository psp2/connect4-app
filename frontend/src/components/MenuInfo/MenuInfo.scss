.Tutorial {
  position: absolute;
  margin: auto;
  top: 20vh;
  left: 20vw;
  right: 20vw;
  z-index: 1;
  height: 30vw;
  width: 30vw;
  border-radius: 20px;
  background-color: white;
  display: flex;
  flex-direction: column;
}

.AboutGame {
  position: absolute;
  margin: auto;
  top: 20vh;
  left: 20vw;
  right: 20vw;
  z-index: 1;
  height: 30vw;
  width: 30vw;
  border-radius: 20px;
  background-color: red;
  display: flex;
  flex-direction: column;
}

.TutorialText {
  margin: auto;
  font-size: 1.7em;
  font-family: "Roboto", sans-serif;
  overflow: scroll;
}

.AboutGameText {
  margin: auto;
  color: white;
  font-size: 1.7em;
  font-family: "Roboto", sans-serif;
}

.CloseTutorial {
  height: 2vw;
  width: 12vw;
  font-family: "Roboto", sans-serif;
  font-size: 1em;
  border-radius: 30px;
  background-color: rgb(15, 196, 15);
  margin: auto;
  margin-bottom: 2vh;
  display: flex;
  flex-direction: column;
}

.CloseAboutGame {
  height: 2vw;
  width: 12vw;
  font-family: "Roboto", sans-serif;
  font-size: 1em;
  border-radius: 30px;
  background-color: white;
  margin: auto;
  margin-bottom: 2vh;
  display: flex;
  flex-direction: column;
}
