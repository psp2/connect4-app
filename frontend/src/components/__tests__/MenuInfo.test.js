import React from "react";
import MenuInfo from "../MenuInfo/MenuInfo";
import renderer from "react-test-renderer";

it("MenuInfo, renders correctly", () => {
  // By default, should show About the Game display
  const tree = renderer.create(<MenuInfo />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("MenuInfo, How to Play", () => {
  let displayType = "tutorial";
  const tree = renderer.create(<MenuInfo displayType={displayType} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("MenuInfo, null", () => {
  // By default, should show About the Game display
  const tree = renderer.create(<MenuInfo displayType={null} />).toJSON();
  expect(tree).toMatchSnapshot();
});
