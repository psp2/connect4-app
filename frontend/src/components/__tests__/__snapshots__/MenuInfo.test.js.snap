// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuInfo, How to Play 1`] = `
<div
  className="Tutorial"
>
  <div
    className="TutorialText"
  >
    Here's how to play Connect 4 ... write more once implementation is complete!
  </div>
  <div
    className="CloseTutorial"
  >
    <div
      className="ButtonCaption"
    >
      Close
    </div>
  </div>
</div>
`;

exports[`MenuInfo, null 1`] = `
<div
  className="AboutGame"
>
  <div
    className="AboutGameText"
  >
    This Connect4 app and implementation was designed by Prashant Pokhriyal and Vivek Sriramineni (based on Python scripts written by Prashant Pokhriyal) for a UIUC CS242 class project.
 
 Connect Four belongs to Hasbro, Inc., and we do not aim to profit off of this game. 

 With that being said, sit back and enjoy the game!
  </div>
  <div
    className="CloseAboutGame"
  >
    <div
      className="ButtonCaption"
    >
      Close
    </div>
  </div>
</div>
`;

exports[`MenuInfo, renders correctly 1`] = `
<div
  className="AboutGame"
>
  <div
    className="AboutGameText"
  >
    This Connect4 app and implementation was designed by Prashant Pokhriyal and Vivek Sriramineni (based on Python scripts written by Prashant Pokhriyal) for a UIUC CS242 class project.
 
 Connect Four belongs to Hasbro, Inc., and we do not aim to profit off of this game. 

 With that being said, sit back and enjoy the game!
  </div>
  <div
    className="CloseAboutGame"
  >
    <div
      className="ButtonCaption"
    >
      Close
    </div>
  </div>
</div>
`;
